### DISCUSSION TITLE
Example to use parameters in a configuration.dsc.yaml file

### DISCUSSION THREAD

**Discussion by FrancisGauthier** [microsoft/winget-cli#4850] [Configuration]
Hi,

Ive been experimenting winget configure using a yaml file to set up work stations for developpers.

I want to use parameters to set up the dev drive letter

here is a part of the actual yaml file with the drive letter hardcoded to "E:"

![image](https://github.com/user-attachments/assets/17ac02b1-f909-42f8-b782-b6505263ce81)

i would like to use a parameter, so a developper could set up the drive letter to "Z:" instead. 

![image](https://github.com/user-attachments/assets/427ad6ef-f576-4b66-b4d5-385a392595d2)

Another benefit would be that we won't have to scan to whole file to find out where the devdrive ressource is in order to change the setting value. 

I saw that there is a "parameters" defined in the winget configure schema file. Are there examples available? I do not see any in the doc neither on microsoft's documentation. Here is the part of the schema that i am referencing to:

                 "parameters": {
                    "type": "array",
                    "items": { "$ref": "#/$defs/resource" },
                    "description": "Resources that retrieve information via a 'get' operation."
                },

Thanks

**Comment by denelon** [comment]
We're looking at the schema for DSC v3 which includes support for parameters. In a future preview release we will include experimental support.

There is an [example](https://github.com/PowerShell/DSC/blob/df2d3abf331d45c87c94a68899dd1c472543fca5/dsc/examples/osinfo_parameters.dsc.yaml#L4) over in the PowerShell DSC project.
